File: web/wrapper.js
/**
 * This file exposes APIs for usage by Dart code in `wrapper.dart`,
 * and serves as the webpack entry point.
 */
import RaisedButton from 'material-ui/RaisedButton';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import * as colors from 'material-ui/styles/colors';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

window.MaterialUiWrapper = {
  // Prefix with underscores so that the Dart getters can be private.
  // Makes the wrapper cleaner.
  _FlatButton: FlatButton,
  _Dialog: Dialog,
  _RaisedButton: RaisedButton,
  _MuiThemeProvider: MuiThemeProvider,

  colors,
  getMuiTheme,
};

File: web/wrapper.dart
/// This file provides Dart wrappers for the Material-UI APIs
/// exposed in `wrapper.js`.
@JS('MaterialUiWrapper')
library wrapper;

import 'package:js/js.dart';
import 'package:react/react_client.dart';
import 'package:react/react_client/react_interop.dart';

//
// Wrapped APIs
//

@JS()
external Colors get colors;

@JS()
external getMuiTheme(config);

@JS()
@anonymous
class Colors {
  external String get deepOrange500;
}

//
// Wrapped React component classes
//

@JS()
external ReactClass get _FlatButton;

@JS()
external ReactClass get _Dialog;

@JS()
external ReactClass get _RaisedButton;

@JS()
external ReactClass get _MuiThemeProvider;

// Create react-dart factories from the existing JS component classes.
final FlatButton = new ReactJsComponentFactoryProxy(_FlatButton);
final Dialog = new ReactJsComponentFactoryProxy(_Dialog);
final RaisedButton = new ReactJsComponentFactoryProxy(_RaisedButton);
final MuiThemeProvider = new ReactJsComponentFactoryProxy(_MuiThemeProvider);
